### ISSUE TITLE
False positive in "duplicate_imports" corner case

### ISSUE THREAD

**Issue by ptsochantaris** [realm/SwiftLint#6098]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

SwiftLint produces a false positive in this case:

![Image](https://github.com/user-attachments/assets/f6a8e8a6-a946-448d-a4e6-6f76d2f5f1b0)

### Environment

* SwiftLint version - *0.59.1*
* Xcode version - *16.4*
* Installation method used - *Homebrew*
* Configuration file:

```yml
disabled_rules:
  - identifier_name
  - line_length
  - trailing_whitespace
  - force_cast
  - file_length
  - function_body_length
  - nesting
  - notification_center_detachment
  - unused_setter_value
  - cyclomatic_complexity
  - type_body_length
  - force_try
  - implicit_getter
  - large_tuple

excluded:
  - Pods

analyzer_rules:
  - unused_import
```

**Comment by SimplyDanny** [comment]
This is actually the intended behavior. The assumption is that when you import `CoreImage`, `CIFilterBuiltins` is already visible.

**Comment by ptsochantaris** [comment]
You know, I had thought that as well, but it turns out in this case it doesn't in this specific weird case! I don't know why, I have a theory though.

The `CIFilterBuiltins` package is specifically imported to support Swift-only strong typing for Core Image. For example, without this import, the type-safe abstractions for CI types don't show up:

<img width="827" alt="Image" src="https://github.com/user-attachments/assets/e18d834c-db17-4a78-8e1c-50f9f9709387" />

So I assume that while `CoreImage` is a module wrapper for the ObjC framework, the `CIFilterBuiltins` is possibly a distinct module in Swift that happens to be "typographically" namespaced under the CI framework, but in actuality isn't a "proper" submodule? Or maybe I'm totally wrong about this :D 

However it's definitely not the case that the single import will bring that other namespace in, in this specific case.

To be fair, this may be such a corner case that you'd be justified in sticking it with a "won't fix" :) but because it's such a corner case I thought I'd report it anyway.

(BTW interesting post related to exactly the weird import issue here: https://mjtsai.com/blog/2025/02/24/cifilterbuiltins/)

**Comment by SimplyDanny** [comment]
Turns out that there are a few of these modules requiring an explicit import. I've put up [a fix](https://github.com/realm/SwiftLint/pull/6203) to exclude them from the analysis.
